### DISCUSSION TITLE
Finding first instance given a threshold (density threshold-based calculation)

### DISCUSSION THREAD

**Discussion by pecos27** [pydata/xarray#5926] [Q&A] [answered]
I have a 3d array, time, depth and longitude (just think a cross section along longitude at one specific latitude).
I am trying to calculate the mixed layer depth based on the potential density criterion (potential density exceeds that at 10m by more than 0.01 kg/m^3). 

The threshold is: the local potential density (referenced to the surface) does not exceed the potential density at 10 m by more than 0.01 kg/m^3. However, in order to avoid interpolation, I would like to find the first instance where the depth exceeds the potential density at 10m + the threshold (.01 kg/m3). 

My goal is to find the density/depth where it exceeds the threshold the first time.
I have the potential density referenced to the surface, and I made the threshold:
```
# sigma_63_N_monthly_10m = potential density at 10m
sigma_63_N_monthly_10m_threshold = sigma_63_N_monthly_10m + .01 #.01 kg/m3 threshold

#sigma_63_N_monthly = local potential density referenced to the surface
sigma_exceeds_threshold = sigma_63_N_monthly.where(sigma_63_N_monthly > sigma_63_N_monthly_10m_threshold) 
```

But when you do `.where > threshold`, it finds all the values which are greater. I would like to just locate the first instance where the density exceeds that threshold and then find the corresponding depth to indicate the mixed layer depth. Something along the lines of finding the index where sigma exceeds the threshold for the first time, and applying that index to get the corresponding depth...  
Is there a way to do that via xarray? 

The data itself is not the most reproducible, but it goes along the lines of something like this. Courtesy of this stackoverflow answer: https://stackoverflow.com/questions/64144035/how-can-i-efficiently-calculate-the-first-instance-of-a-value-in-an-axis-in-dask

```
da = xr.DataArray(np.random.rand(5,5,5), 
                          dims=list('abc'), 
                          coords=dict(c=range(5)))

da.shape

(5, 5, 5)

greater_than_threshold = da.where(lambda x: x > .8)

first_index = greater_than_threshold.idxmax(dim='c') # in my case with the first instance exceeding the threshold, maybe idxmin would be the better option?
``` 

Thank you so much! Any suggestions or comments would be much appreciated :)

**Comment by dcherian** [comment, marked as answer]
I use something like this but `idxmax` or `idxmin` should do the same
```
    thresh = xr.where((np.abs(drho) > 0.03) & (N2 > 1e-5), drho.depth, np.nan)
    mld = thresh.max("depth")
```
